_drafts/2020-03-31-march-2020-roundup.md

```diff
@@ -0,0 +1,55 @@
+* [Unleashing Mayhem over Bluetooth Low Energy](https://asset-group.github.io/disclosures/sweyntooth/)  
+  A deep dive analysis into BLE vulnerabilities into popular BLE chips from Dialog, STMicro, and Cypress chips. 
+* [GCC extensions to the C language](https://embeddedbits.org/gcc-extensions-to-the-c-language/) by Sergio Prado  
+  A great overview of various GCC extensions for C. We'd also like to add one of our personal favorites to the list, GCC's [`__cleanup__` attribute](http://echorand.me/site/notes/articles/c_cleanup/cleanup_attribute_c.html) ([example in JerryScript](https://github.com/jerryscript-project/jerryscript/blob/b9f2b1cf25107b7eb6ff3e7a5a8cd18d8334a1dc/jerry-ext/include/jerryscript-ext/autorelease.impl.h#L22-L34))
+* [Introduction to Trusted Execution Environment and ARM's TrustZone](https://embeddedbits.org/introduction-to-trusted-execution-environment-tee-arm-trustzone/) by Sergio Prado
+* [How to Port a Board for RIOT](https://blog.martine-lenders.eu/riot-board-en.html) by Martine Lenders  
+  RIOT being the [RIOT RTOS](https://riot-os.org/). I felt this was post-worthy because one can never read too much about driver abstraction in large projects.
+* [PlatformIO x Renode Demo - Zephyr Build and Debug on SiFive FE310](https://www.youtube.com/watch?v=EdC3kFZxSzI&feature=emb_logo)  
+  This demo has every technical buzzword that people are excited about. PlatformIO, Renode, Zephyr, RISC-V, SiFive...
+* [Exception(al) Failure - Breaking the STM32F1 Read-Out Protection](https://blog.zapb.de/stm32f1-exceptional-failure/) by Marc Schink & Johannes Obermaier.  
+  A wonderfully technical and professional vulnerability write-up. Source-code is also available.
+* [ARM GCC Inline Assembler Cookbook](http://www.ethernut.de/en/documents/arm-inline-asm.html)  
+  I stumbled across this a couple weeks back and figured I'd throw it in the pile.
+* [Hacking Your ELF For Fun And Profit](https://mgalgs.github.io/2013/05/10/hacking-your-ELF-for-fun-and-profit.html) by Mitch Humpherys  
+  Fascinating post about how to use custom ELF sections to build a plug-in system, with a description about how this applies to the Linux Kernel.
+* [Static analysis in GCC 10](https://developers.redhat.com/blog/2020/03/26/static-analysis-in-gcc-10/) by David Malcolm  
+  We **love** GCC and Clang, and we are always excited to see them have new features.
+* [async/await on Embedded Rust](https://ferrous-systems.com/blog/async-on-embedded/) by Ferrous Systems  
+  Using async/await on an ARM Cortex-M, complete with an [open-source example](https://github.com/ferrous-systems/async-on-embedded/). Definitely worth the read.
+* [Mastering Embedded Linux Series](https://www.thirtythreeforty.net/posts/2020/03/mastering-embedded-linux-part-4-adding-features/) by George Hilliard
+
+## Neat Open Source Projects
+
+* [Baselibc](https://github.com/PetteriAimonen/Baselibc)
```

Comment:
Maybe a one-liner here? tangentially related but not super new, I found https://github.com/ARMmbed/minimal-printf/ recently via grep.app, hadn't seen that repo before.  readable printf impl and pretty low memory footprint!